src/hooks/useTriviaProblems.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback, useState } from "react";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { RootStackParamList } from "../types";
+
+type Props = NativeStackScreenProps<RootStackParamList, "TriviaMain">;
+export default function useTriviaProblems({ navigation, route }: Props) {
+  const [questionsAttempted, setQuestionsAttempted] = useState(0);
+  const [questionsCorrect, setQuestionsCorrect] = useState(0);
+  const [totalTime, setTotalTime] = useState(0);
```

Comment:
Your logic for getting the total time does not work as intended. It currently adds the negative of the total elapsed time since the start, instead of adding the elapsed time since we received the current question. Vidushi pointed out that the avg time statistic can be calculated by `TOTAL_TIME / questionsAttempted`. Use this formula in `onTimeComplete`, and remove all instances of `elapsedTimeRef`. Also, remove the `totalTime` state, and `timeTaken` parameter from the `updateStatsOnAnswer` function.